src/lib.rs

```diff
@@ -53,6 +60,29 @@ pub trait TcpStack {
 	fn close(&self, socket: Self::TcpSocket) -> Result<(), Self::Error>;
 }
 
+/// This trait is implemented by TCP/IP stacks that expose TCP server functionality. TCP servers
+/// may listen for connection requests to establish multiple unique TCP connections with various
+/// clients.
+pub trait TcpServer: TcpClient {
+	/// Create a new TCP socket and bind it to the specified local port.
+	///
+	///Returns `Ok(socket)` when a new socket is successfully created and bound to the specified
+	///local port. Otherwise, an `Err(e)` variant is returned.
+	fn bind(&self, local_port: u16) -> Result<Self::TcpSocket, Self::Error>;
```

Comment:
Why does `bind()` create and return a socket, but `connect()` _accepts_ a socket?  Is that intentional?

Reply:
I have updated all sockets to be created using `socket()` and then updated all other functions to accept a reference to the original socket.